### DISCUSSION TITLE
FEATURE Request - Add Label/Value output filter for Simple Custom Defined Lists in Pods Templates

### DISCUSSION THREAD

**Discussion by jimtrue** [pods-framework/pods#6619] [Ideas]
We should probably have a built-in display output filter that makes it easier for users to display either the label or value for Simple Custom Defined Lists in Pods Templates with the Magic Tags.

Default: `{@simple_list}` will always display the value; we need a `{@simple_list,label}`. Might be useful for other fields as well.

**Comment by sc0ttkclark** [comment]
`{@simple_list}` should always display the label in a `value|Label` list. If that's not the case, this is a bug.

**Comment by jimtrue** [comment]
Then we're looking for the opposite handler, value

**Comment by frank-cruz-f** [comment]
Hello guys, since I think this is currently supported <code>{@simple_list, value}</code> this should be closed,
Thanks

**Comment by quasel** [comment]
Did test myself 
{@simple_listl}  -> label
{@simple_listl, not_existing_function} -> value 
which makes perfectly sense...

**Comment by jimtrue** [comment]
Just did the same test, I'm only getting the 'label' in this case, whether I use nothing or use {@simple_list, value} or use {@simple_list, not_existing_function}. 

However, in my case the Simple List had **multiple** selections. When I changed it to just one, I get the Value (though why **not_existing_function** should work makes absolutely no sense, whatsoever).

So if 'value' is supposed to work, it's not being called when the display function in Pods Template is called against the Simple List Relationship when it has more than one record.

**Comment by Shelob9** [comment]
That actually makes sense. In the `{@tag,callback}` syntax it passed the
raw value (IE not the label in this context) to the callback. Pods must be
testing if the callback exists (as to avoid a fatal error) and if it
doesn't, it just returns.

## 

JoshPress.net http://joshpress.net/
CalderaWP.com http://calderawp.com/
@Josh412 http://twitter.com/Josh412

**Comment by jimtrue** [comment]
Sounds like we need an 'actual' handler, then, correct? One that also properly processes if a list/array is passed to the value function. Most Simple Lists that have come through the forums are Multiple Selections, which is probably why this hasn't worked.

**Comment by quasel** [comment]
well this is a Design decision ^^ wether to change the default return for an array() to a comma seperated list or if to implement a pods_value() sort of function as it
 is done with pods_video - http://pods.io/docs/build/using-magic-tags/

**Comment by m7ess5** [comment]
is this feature still under development ?

**Comment by sc0ttkclark** [comment]
Not currently being worked on, no.

**Comment by Aetles** [comment]
Not that this is being worked on anyway, but it would be useful to get the label for any field, not just lists.

**Comment by jimtrue** [comment]
@Aetles That's a completely different request. Calling it a 'label' is the confusion. In Simple Custom Defined Lists, you enter a set of values that are input as "Label|Value". That's not the same as the "Label" for the field. If you would like to add an enhancement request for that, please create a new issue. Thanks!

**Comment by Aetles** [comment]
Ah, I see. Thanks for the clarification!

**Comment by jimtrue** [comment]
For additional clarification, we still need this, because we have no way to get the 'label' in an `[each simple_list][/each]` loop. You can get the 'value', using `{@_value}` inside the each loop, but nothing I'm running across provides the 'label'.

**Comment by douglsmith** [comment]
FYI, When adding custom defined options for a field, the help text shows it to be swapped from what you said above, @jimtrue. The help says it should be _value|Label_. The _Label_ is what is displayed to the user in radio buttons, checkboxes, and lists. The _value_ is what is passed when submitting the form.

`{@field}` outputs the Label, as I would expect since that is the human readable text.
`{@field, value}` outputs the value.

These are both working for me in an `[each][/each]` loop with a simple custom defined list, multiple select field.

**Comment by jimtrue** [comment]
@douglsmith Sorry just now replying on this one as I'd like to close it if it's actually supported.

> These are both working for me in an [each][/each] loop with a simple custom defined list, multiple select field.

How are you referencing these fields within an `[each field][/each]` if `field` is the simple custom defined list? What are you using within the `each` block to reference the 'label' and the 'value'?

**Comment by douglsmith** [comment]
I hope I wasn't talking about something different. Here's an example of what I was doing.

I had a _Supplies_ custom post type with several fields. One of those fields specified the color of the supply. That field was a simple custom defined list. I wanted the user to pick a simple color name like _Red_ or _Blue_ but I wanted to store the color's hex code to later display as a swatch on the screen.

Here's the field definition …

<img width="524" alt="screen shot 2018-02-26 at 3 25 48 pm" src="https://user-images.githubusercontent.com/2446310/36696310-7589f728-1b09-11e8-8f11-28e77396f42c.PNG">

Which displays like this when editing a post …

<img width="616" alt="screen shot 2018-02-26 at 3 27 15 pm" src="https://user-images.githubusercontent.com/2446310/36696388-a5fd9a40-1b09-11e8-8980-4f0146ff4207.PNG">

A Pod template then displayed the records like this. It set the title of a div to the user-friendly color name (the label), but set background-color in the style to the hex code (the value).

```
[each supplies]
	[if supply_color]
		<div class="swatch" title="{@supply_color}" style="background-color:{@supply_color,value}"></div>
	[/if]
[/each]
```

And that outputs something like this in the HTML for each record …

```<div class="swatch" title="Blue" style="background-color:#0000ff"></div>```

(Disclaimer for those who may stumble across this later: This was just for testing a while back and is not practical for very many color values. I've abandoned it in favor of separate color name and color picker fields. Those fields are then output as a title and color value, which changes the color of a cool looking SVG symbol.)

**Comment by m7ess5** [comment]
maybe call it "field_label" instead of "label". the ability to display the field label function is needed in a lot of cases. please see attachment for clarifications:
![labels](https://user-images.githubusercontent.com/1860808/50745141-aaa9a500-127c-11e9-8453-9cb9ea61e141.jpeg)
the result of
`{@ticket_filename,field_label}` 
should be: Admission

**Comment by raoulunger** [comment]
Stumbling upon here in 2020: has this been taken up any further? I have a pod with a field of type 'relationship', related to a 'simple (custom defined list)'. I allow for multiple (2) selections, and have the values come up through a magic tag. The options in the list have a value and a label like this: 

b-1-3|1/3

This displays fine in the pod list (as 1/3). But the magic tag returns the label, not the value.

I've seen solutions suggested such as {@my-field,value}, and that works fine with a single selection, but not with multiple ones (as mentioned above). Any developments towards a solution?

Thanks!

**Comment by JoryHogeveen** [comment]
@maustyle 
That is a whole different feature. Label is used for all fields.

@raoulunger 
From what I read in @douglsmith posts multiple values should work fine with an `[each]` loop.
Other than that, as this topic was closed, no there is currently no dev going on for this.
